src/Domain/FunctionsOfTime/QuaternionHelpers.cpp

```diff
@@ -7,7 +7,6 @@
 
 #include "DataStructures/DataVector.hpp"
 #include "Utilities/ErrorHandling/Assert.hpp"
-#include "Utilities/Gsl.hpp"
```

Comment:
You have to keep the include in the cpp file, because the type must be complete for the function definition, as @wthrowe said above :)